scripts/capture-marketing-external.sh

```
@@ -0,0 +1,150 @@
+# $1 marker dir, $2 step name, $3 destination png, $4 capture command
+drive_step() {
+	local dir="$1" name="$2" dest="$3"; shift 3
+	local marker="$dir/ready-$name" waited=0
+	while [ ! -f "$marker" ]; do
+		sleep 0.5; waited=$((waited + 1))
+		if [ "$waited" -gt 240 ]; then echo "    ✗ timed out waiting for $name" >&2; return 1; fi
+	done
+	sleep 1.5   # let the last frame present after the app says it is ready
+	"$@" "$dest"
+	rm -f "$marker"
+	[ -f "$dest" ] && echo "    ✓ $name" || { echo "    ✗ capture failed: $name" >&2; return 1; }
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)